framework/src/controller/application.js

```diff
@@ -66,76 +66,56 @@ class Application {
 	 * @param {Object} genesisBlock - Genesis block object
-	 * @param {Object} [constantsToOverride] - Override constantsToOverride
-	 * @param {Object|Array.<Object>} [config] - Main configuration object or the array of objects, array format will facilitate user to not deep merge the objects
+	 * @param {Object} [config] - Main configuration object
+	 * @param {string} [config.app.label] - Label of the application
+	 * @param {Object} [config.app.genesisConfig] - Configuration for applicationState
+	 * @param {string} [config.app.version] - Version of the application
+	 * @param {string} [config.app.minVersion] - Minimum compatible version on the network
+	 * @param {string} [config.app.protocolVersion] - Compatible protocol version application is using
 	 * @param {Object} [config.components] - Configurations for components
 	 * @param {Object} [config.components.logger] - Configuration for logger component
 	 * @param {Object} [config.components.cache] - Configuration for cache component
 	 * @param {Object} [config.components.storage] - Configuration for storage component
-	 * @param {Object} [config.initialState] - Configuration for applicationState
 	 * @param {Object} [config.modules] - Configurations for modules
-	 * @param {string} [config.version] - Version of the application
-	 * @param {string} [config.minVersion] - Minimum compatible version on the network
-	 * @param {string} [config.protocolVersion] - Compatible protocol version application is using
 	 *
 	 * @throws Framework.errors.SchemaValidationError
 	 */
-	constructor(
-		label,
-		genesisBlock,
-		config = { app: {}, components: { logger: null }, modules: {} }
-	) {
-		let appConfig;
-
-		// If user passes multiple config objects merge them in left-right order
-		if (Array.isArray(config)) {
-			// We don't have a mergeDeep method, so we are using defaultsDeep
-			// in the reverse order to have same behaviour
-			appConfig = _.defaultsDeep(...config.reverse());
-		} else {
-			appConfig = config;
-		}
+	constructor(genesisBlock, config = {}) {
+		validator.validate(genesisBlockSchema, genesisBlock);
 
-		if (!appConfig.components.logger) {
-			appConfig.components.logger = {
-				logFileName: `${process.cwd()}/logs/${label}/lisk.log`,
-			};
-		}
+		// Don't change the object parameters provided
+		let appConfig = _.cloneDeep(config);
 
-		validator.loadSchema(constantsSchema);
-		validator.loadSchema(applicationSchema);
+		if (!_.has(appConfig, 'app.label')) {
+			_.set(appConfig, 'app.label', genesisBlock.payloadHash);
+		}
 
-		// If app label is a function it will be dependent on compiled configuration
-		// so we assign and validate it later stage
-		if (typeof label === 'string') {
-			validator.validate(applicationSchema.appLabel, label);
+		if (!_.has(appConfig, 'components.logger.logFileName')) {
+			_.set(
+				appConfig,
+				'components.logger.logFileName',
+				`${process.cwd()}/logs/${appConfig.app.label}/lisk.log`
```

Comment:
If you are referring `generate_configuration` script yes its per network as its part of the Lisk Core. 

If you are referring to `getConfig` method that is independent of any network. You can load as many configuration on top of each other as a stack and it will merge them all with defaults and gave you final config.